sbt-plugin-test/testFramework/src/main/scala/sbttest/framework/DummyTask.scala

```diff
@@ -34,7 +36,8 @@ final class DummyTask(
 
   def execute(eventHandler: EventHandler, loggers: Array[Logger],
       continuation: Array[Task] => Unit): Unit = {
-    continuation(execute(eventHandler, loggers))
+    val tasks = execute(eventHandler, loggers)
+    runner.taskBlock.foreach(_ => continuation(tasks))
```

Comment:
This is specific to the dummy framework. It tests that if a framework relies on receiving messages from the master to run tests, it can wait on the messages.